includes/class-newspack-popups.php

```diff
@@ -988,19 +1017,22 @@ public static function duplicate_popup( $id, $title = '' ) {
 		$new_popup_id = false;
 
 		if ( is_a( $old_popup, 'WP_Post' ) && self::NEWSPACK_POPUPS_CPT === $old_popup->post_type ) {
-			$duplicate_of = get_post_meta( $id, 'duplicate_of', true );
-			$original_id  = 0 < $duplicate_of ? $duplicate_of : $id; // If the post we're duplicating is itself a copy, inherit the 'duplicate_of' value. Otherwise, set the value to the post we're duplicating.
-			$new_popup    = [
+			$duplicate_of        = get_post_meta( $id, 'duplicate_of', true );
+			$original_id         = 0 < $duplicate_of ? $duplicate_of : $id; // If the post we're duplicating is itself a copy, inherit the 'duplicate_of' value. Otherwise, set the value to the post we're duplicating.
+			$original_title_base = self::get_duplicated_post_base_title( get_the_title( $original_id ), $title );
+			$new_popup           = [
 				'post_type'     => self::NEWSPACK_POPUPS_CPT,
 				'post_status'   => 'draft',
-				'post_title'    => ! empty( $title ) ? $title : self::get_duplicate_title( $original_id ),
+				'post_title'    => ! empty( $title ) ? $title : self::get_duplicate_title( $duplicate_of, $id ),
 				'post_author'   => $old_popup->post_author,
 				'post_content'  => $old_popup->post_content,
 				'post_excerpt'  => $old_popup->post_excerpt,
 				'post_category' => wp_get_post_categories( $id, [ 'fields' => 'ids' ] ),
 				'tags_input'    => wp_get_post_tags( $id, [ 'fields' => 'ids' ] ),
 				'meta_input'    => [
-					'duplicate_of' => $original_id,
+					// A campaign is set as the origin of another one, if the later have the same title with the count of occurences suffixed (e.g. my prompt 3).
+					'duplicate_of' => empty( $title ) || preg_match( "/^$original_title_base\s*\d*$/", $title )
```

Comment:
If we're solely basing the copy titles off of the name of the prompt being copied (whether or not it is itself a duplicate) plus the existence of other prompts with the same title, I believe we might be able to deprecate the `duplicate_of` meta field, right?